invenio_app_ils/records/jsonresolver/document.py

```diff
@@ -27,36 +24,23 @@ def jsonresolver_loader(url_map):
     def circulation_resolver(document_pid):
         """Return circulation info for the given Document."""
         circulation = {}
-        active_loans_count = 0
-        loanable_items_count = 0
-        items_count = 0
-        item_pids = circulation_items_retriever(document_pid)
-
-        for pid in item_pids:
-            items_count += 1
-            if not is_item_available(pid):
-                active_loans_count += 1
 
-            if circulation_is_item_available(pid):
-                loanable_items_count += 1
+        items = list(ItemSearch().search_by_document_pid(document_pid).scan())
+        items_count = len(items)
+        loanable_items_count = sum(item["status"] == "LOANABLE" for item in items)
 
-        search = search_by_pid(
-            document_pid=document_pid,
-            filter_states=current_app.config.get(
-                "CIRCULATION_STATES_LOAN_COMPLETED", []
-            ),
-        )
+        loan_search = IlsLoansSearch()
 
-        pending_loans = search_by_pid(document_pid=document_pid,
-                                      filter_states=['PENDING']
-                                      ).execute().hits.total
+        past_loans_count = len(list(loan_search.get_past_loans_by_doc_pid(document_pid)))
```

Comment:
You are going to hate me, and I am sorry, but it came to my mind only now. We can even improve this more by avoiding `list` and then `len`. We can ask directly to ES the count.
See [here](https://github.com/inveniosoftware/invenio-circulation/blob/792f5ae7352b7979717c8087e142f3eabe70d0e8/invenio_circulation/search/api.py#L52) and [here](https://github.com/inveniosoftware/invenio-circulation/blob/792f5ae7352b7979717c8087e142f3eabe70d0e8/invenio_circulation/api.py#L69) an example.

Basically we "prepare" the search, but it is when we use it we decide to use `execute` or `scan`.
It should not be a big effort to change it... Sorry...